packages/taquito/src/contract/contract-methods/contract-method-object-param.ts

```diff
@@ -1,80 +1,93 @@
-import { DEFAULT_SMART_CONTRACT_METHOD_NAME } from '../contract';
+import type { Wallet } from '../../wallet/wallet';
+import { TransactionWalletOperation } from '../../wallet/transaction-operation';
+import { ParameterSchema } from '@taquito/michelson-encoder';
+import {
+  ContractMethodInterface,
+  ExplicitTransferParams,
+  SendParams,
+} from './contract-method-interface';
+import { DEFAULT_SMART_CONTRACT_METHOD_NAME } from '../constants';
 
 /**
  * @description Utility class to send smart contract operation
  * The format for the arguments is the object representation
  */
-export class ContractMethodObject<T extends ContractProvider | Wallet> implements ContractMethodInterface {
-    constructor(
-        private provider: T,
-        private address: string,
-        private parameterSchema: ParameterSchema,
-        private name: string,
-        private args: any = 'unit',
-        private isMultipleEntrypoint = true,
-        private isAnonymous = false
-    ) { }
+export class ContractMethodObject<T extends ContractProvider | Wallet>
+  implements ContractMethodInterface
+{
+  constructor(
+    private provider: T,
+    private address: string,
+    private parameterSchema: ParameterSchema,
+    private name: string,
+    private args: any = 'unit',
+    private isMultipleEntrypoint = true,
+    private isAnonymous = false
+  ) {}
 
-    /**
-     * @description Get the signature of the smart contract method
-     */
-    getSignature() {
-        return this.isAnonymous
-            ? this.parameterSchema.ExtractSchema()[this.name]
-            : this.parameterSchema.ExtractSchema();
+  /**
+   * @description Get the signature of the smart contract method
+   */
+  getSignature() {
+    const generatedSchema = this.parameterSchema.generateSchema();
+    if (this.isAnonymous) {
+      const schemaObj = generatedSchema.schema as Record<string, unknown> | undefined;
+      return schemaObj?.[this.name];
     }
+    return generatedSchema;
+  }
 
-    /**
-     *
-     * @description Send the smart contract operation
-     *
-     * @param Options generic operation parameter
-     */
-    send(
-        params: Partial<SendParams> = {}
-    ): Promise<T extends Wallet ? TransactionWalletOperation : TransactionOperation> {
-        if (this.provider instanceof Wallet) {
-            return (this.provider as unknown as Wallet).transfer(this.toTransferParams(params)).send() as any;
-        } else {
-            return this.provider.transfer(this.toTransferParams(params)) as any;
-        }
+  /**
+   *
+   * @description Send the smart contract operation
+   *
+   * @param Options generic operation parameter
+   */
+  send(
+    params: Partial<SendParams> = {}
+  ): Promise<T extends Wallet ? TransactionWalletOperation : TransactionOperation> {
+    // provider is either ContractProvider | Wallet but pkh method is unique to Wallet
+    if (this.provider && 'pkh' in this.provider) {
```

Comment:
Hey @maxwellward, i realized this logic is not shared  in other files anymore so decided to save the trouble of isolating it into a reusable function and risk circular dependencies hence wallet-guard file is removed. And I came to senses to appreciate your original logic which is very neat!